src/View/SSViewer_Scope.php

```diff
@@ -51,44 +56,88 @@ class SSViewer_Scope
      */
     protected $itemIterator;
 
-    //Total number of items in the iterator
+    /**
+     * Total number of items in the iterator
+     *
+     * @var int
+     */
     protected $itemIteratorTotal;
 
-    // A pointer into the item stack for which item should be scope on the next pop call
+    /**
+     * A pointer into the item stack for the item that will become the active scope on the next pop call
+     *
+     * @var int
+     */
     private $popIndex;
 
-    // A pointer into the item stack for which item is "up" from this one
-    private $upIndex = null;
+    /**
+     * A pointer into the item stack for which item is "up" from this one
+     *
+     * @var int
+     */
+    private $upIndex;
+
+    /**
+     * A pointer into the item stack for which the active item (or null if not in stack yet)
+     *
+     * @var int
+     */
+    private $currentIndex;
 
-    // A pointer into the item stack for which item is this one (or null if not in stack yet)
-    private $currentIndex = null;
+    /**
+     * A store of copies of the main item stack, so it's preserved during a lookup from local scope
+     * (which may push/pop items to/from the main item stack)
+     *
+     * @var array
+     */
+    private $localStack = [];
 
-    private $localIndex;
+    /**
+     * The index of the current item in the main item stack, so we know where to restore the scope
+     * stored in $localStack.
+     *
+     * @var int
+     */
+    private $localIndex = 0;
 
-    public function __construct($item, $inheritedScope = null)
+    /**
+     * @var object $item
+     * @var SSViewer_Scope $inheritedScope
+     */
+    public function __construct($item, SSViewer_Scope $inheritedScope = null)
```

Comment:
Typehinting added here, not part of our “public” API but happy to undo this if others disagree